src/providers/EtherspotUiContextProvider.tsx

```diff
@@ -137,10 +138,63 @@ const EtherspotUiContextProvider = ({ children, chainId = 1 }: EtherspotUiContex
     }));
   }
 
+  /**
+   * @name smartWalletAddresses
+   * @description Accesses the underlying Etherspot SDK
+   * and returns the smart wallet address for each chain.
+   *
+   * @returns Array[ISmartWalletAddress]
+   */
+  const smartWalletAddresses = async (): Promise<(ISmartWalletAddress[])> => {
+    // Lets set our definitions
+    let chainIds: (number | undefined)[] = [];
+    const computedContractResponses: ISmartWalletAddress[] = [];
+
+    // Next, let's get our grouped batches
+    const groupedBatches = Object.values<IBatches>(groupedBatchesPerId);
+
+    // Next, loop through the grouped batches, if any...
+    for (let index = 0; index < groupedBatches.length; index++) {
+      // And collect all the data we need about our batches.
+      const groupedBatch = groupedBatches[index];
+      const batches = (groupedBatch.batches ?? []) as IBatch[];
+
+      // Collect the chain IDs used here...
+      chainIds = batches.map((batch) => batch.chainId || chainId);
+    }
+
+    // A little housekeeping - clean up any duplicates
+    chainIds = uniq(chainIds);
+
+    // Our next step is to cycle through all our chain IDs found
+    // in each batch and call the corresponding Etherspot SDK
+    // and computeContractAccount method which will return the
+    // smart wallet address that users can use.
+    for (let index = 0; index < chainIds.length; index++) {
```

Comment:
since this not needs to be sequential u can do async approach with `const addresses = Promise.all(chainIds.map(async (chainid) => { ...`

Reply:
Updated as per our conversation